s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -110,6 +111,13 @@ func (t *TidbClient) GetBucket(ctx context.Context, bucketName string) (bucket *
 		err = handleDBError(err)
 		return
 	}
+	//get versioning for the bucket
+	versionOpts,versionErr := t.GetBucketVersioning(ctx, tmp.Name)
+	if versionErr != nil {
+		return
+	}
+	tmp.Versioning = &pb.BucketVersioning{}
```

Comment:
Only status needed, why use a new struct, BucketVersioning, here?

Reply:
Earlier versioning was just a boolean value, yes or no. but we need different details for example time stamp and versioning info, hence created another struct for that

Reply:
Time stamp and versioning info are for object, right? But what you created is only for bucket.

Reply:
Okay will update it after RC1, we need to think more over versioning struct, whether we need few more details to be added, for example AWS has "MFA delete" attribute also apart from "Status", so if we also need some attribute alike AWS , we can modify versioning struct

Reply:
Is MFA related to versioning? What kind of attribute may need to be added for versioning?

Reply:
YEs MFA is related to versioning, and we can have delete Marker added as an attribute in near future